docs/how-tos/how-to-use-searchkick-for-search-autocomplete.mdx

```diff
@@ -0,0 +1,285 @@
+  in a special way that allows a partial match with the initial
+  part of the string. By default, it matches the entire word.
+  We will see later why we need this.
+- `search_data` tells Searchkick which fields we want to index.
+  For this example, `name` is enough but you could consider adding
+  more fields depending on your needs.
+- `should_index?` determines which records will be indexed. In this
+  case, to keep it simple we only index products that are not
+  soft-deleted.
+
+Now that we've got the basis, it's time for our first real indexing.
+Opening a Rails console, we can run:
+
+```ruby title="bin/rails c"
+Spree::Product.reindex
+```
+
+:::info
+
+Be sure you have Elasticsearch up and running. Please, follow the
+official [installation instructions][elasticsearch-install-guide],
+or if you are a MacOS and Homebrew user, just run:
+
+```sh
+brew install elastic/tap/elasticsearch-full
+brew services start elasticsearch-full
+```
+
+:::
+
+At this poing we can verify that a basic search is working.
+Assuming you have some products which include the `ruby`
+word in one of the indexed fields, you can run:
+
+```ruby title="bin/rails c"
+Spree::Product.search("ruby").map(&:name)
+# => ["Ruby Hoodie", "Ruby Polo", "Ruby Mug", "Ruby Tote", "Ruby Hoodie Zip"]
+```
+
+Yey, it works! 🎉
+
+Time to index our categories as well:
+
+```ruby title="app/overrides/my_store/spree/taxon/add_searchkick.rb"
+module AmazingStore
+  module Spree
+    module Taxon
+      module AddSearchkick
+        def self.prepended(base)
+          unless base.respond_to?(:searchkick_index)
+            base.searchkick word_start: [:name]
+          end
+        end
+
+        def search_data
+          { name: name }
+        end
+
+        ::Spree::Taxon.prepend self
+      end
+    end
+  end
+end
+```
+
+We can now run a reindex for `Spree::Taxon` and verify it's
+working:
+
+```ruby title="bin/rails c"
+Spree::Taxon.reindex
+Spree::Taxon.search("ruby").map(&:name)
+# => ["Ruby"]
+```
+
+## Enhancing search autocomplete
+
+[Solidus Starter Frontend][ssf] comes with a simple search autocomplete out of
+the box. In this section of the guide we will change its behavior to use the
+newly added advanced search functionalities.
+
+:::info
+
+The search autocomplete feature has been added to the Starter Frontend
+since Solidus 3.4.0. If you don't have it, please refer to
+[Autocomplete main search with products and taxons][autocomplete-pr] to
+backport that feature in your storefront.
+
+:::
+
+To let the autocomplete controller use Searchkick, we need to change the
+code of the `AutocompleteResultsController` provided like this:
+
+```diff title="app/controllers/autocomplete_results_controller.rb"
+
+  def autocomplete_products
+    if params[:keywords].present?
+-      searcher = build_searcher(params.merge(per_page: 5))
+-      searcher.retrieve_products
++      Spree::Product.search(params[:keywords], fields: [{name: :word_start}], limit: 5)
```

Comment:
Do we need the `fields` argument?

Reply:
Yes, we need it to tell Searchkick to use the "special" word_start index created before in the model. I will add this information after the code block.